### PR TITLE
Render multi zones in render

### PR DESCRIPTION
**Pull request by neoreact32**
Render multi zones in render

### CODE REVIEW

File: src/neoreact/index.tsx [hunk 1]
```diff
@@ -30,39 +32,48 @@ export class NeoReact<T> implements INeoReact<T> {
...
-
-            this.renderer(comp, el);
-          } catch (err) {}
-        });
+          }
+        };
       }
     }
+
+    Object.keys(renderByService).forEach(serviceName => {
```
Comment (jakejarrett) [inline src/neoreact/index.tsx]: `for (const serviceName  of renderByService) {}` instead please, faster & the variable assigned is immutable.

File: src/neoreact/index.tsx [hunk 2]
```diff
@@ -30,39 +32,48 @@ export class NeoReact<T> implements INeoReact<T> {
...
+      const zoneValue = Object.values(renderByService[serviceName]);
+      let comp: JSX.Element[] = [];
+      const els = document.querySelector(serviceName);
+      zoneValue.forEach((zone, index) => {
```
Comment (jakejarrett) [inline src/neoreact/index.tsx]: Not using index here so can use a for loop instead?

File: src/neoreact/index.tsx [hunk 3]
```diff
@@ -30,39 +32,48 @@ export class NeoReact<T> implements INeoReact<T> {
...
+        if (typeof zone.component === "function") {
+          comp.push(zone.component({}));
+        } else {
+          comp.push(zone.component);
+        }
+      });
+      this.renderer(<React.Fragment>{comp}</React.Fragment>, els);
```
Comment (jakejarrett) [inline src/neoreact/index.tsx]: `this.renderer(comp, els)`?

Also this doesn't support rendering to multiple els anymore.

File: src/neoreact/index.tsx (hunk 1 above)
Comment (neoreact32) [reply src/neoreact/index.tsx]: `renderByService` is not an array.

Comment (neoreact32) [reply src/neoreact/index.tsx]: this should work
```
for (const service in renderByService) {
}
```

Comment (neoreact32) [reply src/neoreact/index.tsx]: done 👍

File: src/neoreact/index.tsx (hunk 2 above)
Comment (neoreact32) [reply src/neoreact/index.tsx]: done 👍

**Pull request merged**